bin/yugabyted

```diff
@@ -2283,5 +2481,140 @@ def retry_op(func, timeout=180):
     raise RuntimeError("Failed after retrying operation for {} secs.".format(
         now - start_time))
 
+
+class SetupEnvBasedCredentials(object):
+    def __init__(self):
+        self.ysql_user = os.environ.get('YSQL_USER')
+        self.ysql_password = os.environ.get('YSQL_PASSWORD')
+        self.ysql_db = os.environ.get('YSQL_DB')
+        self.ycql_user = os.environ.get('YCQL_USER')
+        self.ycql_password = os.environ.get('YCQL_PASSWORD')
+        self.ycql_keyspace = os.environ.get('YCQL_KEYSPACE')
+
+    def is_valid(self, var_name):
+        return True if var_name is not None and var_name else False
+
+    def get_ysql_user(self):
+        return self.ysql_user if self.is_valid(self.ysql_user) else DEFAULT_YSQL_USER
+
+    def get_ysql_password(self):
+        return self.ysql_password if self.is_valid(self.ysql_password) else (self.ysql_user if (
+                    self.is_valid(self.ysql_user)) else DEFAULT_YSQL_PASSWORD)
+
+    def get_ysql_db(self):
+        return self.ysql_db if self.is_valid(self.ysql_db) else (self.ysql_user if (
+                    self.is_valid(self.ysql_user)) else DEFAULT_YSQL_DB)
+
+    def get_ycql_user(self):
+        return self.ycql_user if self.is_valid(self.ycql_user) else DEFAULT_YCQL_USER
+
+    def get_ycql_password(self):
+        return self.ycql_password if self.is_valid(self.ycql_password) else (self.ycql_user if (
+                    self.is_valid(self.ycql_user)) else DEFAULT_YCQL_PASSWORD)
+
+    def get_ycql_keyspace(self):
+        return self.ycql_keyspace if self.is_valid(self.ycql_keyspace) else (self.ycql_user if (
+                    self.is_valid(self.ycql_user)) else DEFAULT_YCQL_KEYSPACE)
+
+    def get_ysql_credentials(self, get_default_credentials=False):
+        if get_default_credentials:
+            return DEFAULT_YSQL_USER, DEFAULT_YSQL_PASSWORD, DEFAULT_YSQL_DB
+        else:
+            return self.get_ysql_user(), self.get_ysql_password(), self.get_ysql_db()
+
+    def get_ycql_credentials(self, get_default_credentials=False):
+        if get_default_credentials:
+            return DEFAULT_YCQL_USER, DEFAULT_YCQL_PASSWORD, DEFAULT_YCQL_KEYSPACE
+        else:
+            return self.get_ycql_user(), self.get_ycql_password(), self.get_ycql_keyspace()
+
+    def is_exists(self, var_to_check):
+        return self.is_valid(os.environ.get(var_to_check))
```

Comment:
the pythonic way to do this is `return var_to_check in os.environ`

Reply:
The motto of `is_exists()` is to verify the argument variable is exist or not and non-empty. And return a boolean value.
The suggested one only checks for the variable is exists or not.

Reply:
Another way to do this 
```py
def is_exists(var_name1):
    return True if var_name1 in os.environ and os.environ.get(var_name1) else False
```